frontend/app/templates/apps/edit.hbs

```diff
@@ -0,0 +1,129 @@
+<div class="container">
+  <div class="col-md-8 col-md-offset-2">
+    <div class="panel panel-default">
+      <div class="panel-heading">
+        <h2 class="panel-title">
+          {{#if creatingApp}}
+            Create New App
+          {{else}}
+            {{!-- Unbound so it doesn't update as the user modifies the name --}}
+            Editing {{unbound model.name}}
+          {{/if}}
+        </h2>
+      </div>
+      <div class="panel-body">
+        <form class="form-horizontal">
+          {{!-- TODO: add logo upload button and display logo
+          <div class="form-group">
+            <label for="app-logo" class="col-sm-4 control-label required-label">Logo</label>
+            <div class="col-sm-8">
+              <img src="{{model.logoUrl}}">
+              {{view "file-upload" action="logoSelected" id="app-logo"}}
+            </div>
+          </div>
+          --}}
+
+          <div class="form-group {{if errors.model.name 'has-error'}}">
+            <label for="app-name" class="col-sm-4 control-label required-label">Name</label>
+            <div class="col-sm-8">
+              {{input type="text" class="form-control" id="app-name"
+                      value=model.name placeholder="Application Name"}}
+            </div>
+          </div>
+          {{show-errors if=model.name fieldName="Name" clientErrors=errors.model.name
+                        serverErrors=model.errors.name}}
+
+          <div class="form-group {{if errors.model.description 'has-error'}}">
+            <label for="app-desc" class="col-sm-4 control-label required-label">Description</label>
+            <div class="col-sm-8">
+              {{input type="text" class="form-control col-sm-8" id="app-desc"
+                      value=model.description placeholder="Description"}}
+            </div>
+          </div>
+          {{show-errors if=model.description fieldName="Description"
+                        clientErrors=errors.model.description
+                        serverErrors=model.errors.description}}
```

Comment:
I know what you're thinking: "nuck wtf why do we have separate clientErrors and serverErrors" — the answer is dockyard/ember-validations#122

Once they fix that and we get on 2.0.0, this whole thing becomes a _lot_ simpler.